sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -484,6 +484,35 @@ case class GpuInitCap(child: Expression) extends GpuUnaryExpression with Implici
     input.getBase.toTitle
 }
 
+case class GpuStringRepeat(strs: Expression, repeatTimes: Expression)
+    extends GpuBinaryExpression with ImplicitCastInputTypes {
+  override def left: Expression = strs
+
+  override def right: Expression = repeatTimes
+
+  override def dataType: DataType = strs.dataType
+
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(TypeCollection(StringType, BinaryType), IntegerType)
+
+  def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
+
+  def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
+
+  def doColumnar(strs: GpuColumnVector, repeatTimes: GpuScalar): ColumnVector = {
+    val repeatTimesVal = repeatTimes.getValue.asInstanceOf[Int]
```

Comment:
Rather than use `asInstanceOf` it would be better to use pattern matching so we can throw nicer errors if we get an unexpected type here. Something like:

```scala
repeatTimes.getValue match {
  case repeatTimesVal: Int => strs.getBase.repeatStrings(repeatTimesVal)
  case _ => throw new IllegalStateException(s"Invalid data type for repeatTimes")  
}
```

Reply:
It would be good to show the actual type in the error message as well